pkg/asset/cluster/tfvars/tfvars.go

```diff
@@ -501,11 +495,15 @@ func (t *TerraformVariables) Generate(parents asset.Parents) error {
 		}
 		// Based on the number of workers, we could have the following outcomes:
 		// 1. workers > 0, masters not schedulable, valid cluster
-		// 2. workers = 0, masters schedulable, valid compact cluster but currently unsupported
+		// 2. workers != compute replicas (possibly caused by worker manifests being manipulated after generation), masters schedulable, invalid cluster
 		// 3. workers = 0, masters not schedulable, invalid cluster
-		if len(workers) == 0 {
-			return errors.Errorf("compact clusters with 0 workers are not supported at this time")
+		if int64(len(workers)) != numWorkers {
```

Comment:
We are comparing different things here:
* `len(workers)` == the number of worker machinesets
* `numWorkers` == the number of replicas in all worker machinesets.

Reply:
As I suspected, GCP installs are broken:
```
level=info msg=Credentials loaded from environment variable "GOOGLE_CLOUD_KEYFILE_JSON", file "/var/run/secrets/ci.openshift.io/cluster-profile/gce.json"
level=fatal msg=failed to fetch Terraform Variables: failed to generate asset "Terraform Variables": number of worker assets: 4 does not match requested workers: 3
Installer exit with code 1
```
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8370/pull-ci-openshift-installer-master-e2e-gcp-ovn/1788264305544663040

Reply:
I didn't realize that the number of worker manifests == number of AZs that the cluster is going to be installer in.

numWorkers which is the sum of compute replicas across all worker manifests is a better metric for our checks. Updated fix to incorporate this understanding.